### ISSUE TITLE
`mnr_submitBitcoinBlock` failed because of `Expect: 100-continue`

### ISSUE THREAD

**Issue by YihaoPeng** [rsksmart/rskj#494]
When the body large than 1024 bytes, `curl` or `libcurl` will send a `Expect: 100-continue` header.
> References:
> * [RFC2616 - Use of the 100 (Continue) Status](https://www.w3.org/Protocols/rfc2616/rfc2616-sec8.html#sec8.2.3)
> * ['Expect: 100-Continue' Issues and Risks](https://support.urbanairship.com/hc/en-us/articles/213492003--Expect-100-Continue-Issues-and-Risks)

It seems like:

```http
Host: 127.0.0.1:4444
Authorization: Basic ......
User-Agent: curl
Accept: */*
Content-Type: application/json
Content-Length: 15224
Expect: 100-continue

```

Then `rskj` will throw a exception:
```log
java.lang.UnsupportedOperationException: unsupported message type: DefaultFullHttpResponse (expect
ed: ByteBuf, FileRegion)
        at io.netty.channel.nio.AbstractNioByteChannel.filterOutboundMessage(AbstractNioByteChanne
l.java:281)
        at io.netty.channel.AbstractChannel$AbstractUnsafe.write(AbstractChannel.java:707)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.write(DefaultChannelPipeline.java:1
122)
        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContex
t.java:633)
        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java
:691)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerCont
ext.java:681)
        at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerCont
ext.java:716)
        at io.netty.handler.codec.http.HttpObjectAggregator.decode(HttpObjectAggregator.java:132)
        at io.netty.handler.codec.http.HttpObjectAggregator.decode(HttpObjectAggregator.java:55)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java
:89)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandler
Context.java:308)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerCo
ntext.java:294)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:244)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandler
Context.java:308)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerCo
ntext.java:294)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:846
)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.j
ava:131)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
```

And the request will failed without any response:
```log
E0227 00:45:16.244127 12213 Utils.cc:216] unable to request data from: http://127.0.0.1:4444, error: Server returned nothing (no headers, no data)
```

----------------------

I have to force `libcurl` switches to HTTP/1.0 to avoid the issue: https://github.com/btccom/btcpool/commit/63e5f9fd3f158fd7778b147841cfc7d69c258ef3

---------------------

```bash
java -version                                                                                                                     !6870
openjdk version "1.8.0_151"
OpenJDK Runtime Environment (build 1.8.0_151-8u151-b12-0ubuntu0.16.04.2-b12)
OpenJDK 64-Bit Server VM (build 25.151-b12, mixed mode)
```

**Comment by martinmedina** [comment]
This is a known issue that has been fixed on master but not included in the last release (0.4.0).
A new release with this fix is coming out in the next days. 

In the meantime you can use [fix_expect_header](https://github.com/rsksmart/rskj/tree/fix_expect_header) branch that includes a patch for the issue.

This new branch will also allow you to revert this [btccom/btcpool@63e5f9f](https://github.com/btccom/btcpool/commit/63e5f9fd3f158fd7778b147841cfc7d69c258ef3)

Let me know if this helps!

**Comment by YihaoPeng** [comment]
That's great. Thank you.
